redash/query_runner/pg.py

```diff
@@ -63,6 +63,38 @@ def _wait(conn, timeout=None):
             raise psycopg2.OperationalError("select.error received")
 
 
+def full_table_name(schema, name):
+    if '.' in name: 
+        name = u'"{}"'.format(name)
+    
+    return u'{}.{}'.format(schema, name)
+
+
+def build_schema(query_result, schema):
+    # By default we omit the public schema name from the table name. But there are
+    # edge cases, where this might cause conflicts. For example:
+    # * We have a schema named "main" with table "users".
+    # * We have a table named "main.users" in the public schema.
+    # (while this feels unlikely, this actually happened)
+    # In this case if we omit the schema name for the public table, we will have
+    # a conflict.
+    table_names = set(map(lambda r: full_table_name(r['table_schema'], r['table_name']), query_result['rows']))
```

Comment:
IIRC, as part of the Python 3 PR, there's a massive conversion of maps to comprehensions. While we can argue on which is nicer (hey, I'm a map guy!), I do feel should consider a consolidation of styles.

Reply:
@NicolasLM is this something `2to3` does on its own? 

Until `python-3` hits master branch and as we merge new code, I would like to reduce the amount of manual changes you need to do. So if there are anything like this that you need to manually update, please let us know so we are mindful of it.

Reply:
Yes, 2to3 replaces maps with list comprehensions (but only when the first argument given to map is a lambda). This is because in Python 3 `map` returns a lazy iterator while in Python 2.7 it returns a list, so 2to3 tries to preserve the return type. A similar thing happens with `dict.keys()` or `dict.values()`.

In this specific example it does not matter since everything is wrapped in a set just after, so it will work in both Python 2.7 and Python 3.

I don't think you need to bother too much about this until the `python-3` branch is merged as these incompatibilities are usually caught early on in tests after a rebase.